### PR TITLE
implement (leak for now unconfirmed) 4 cards from MH3

### PR DESCRIPTION
**Pull request by Susucre**
So there are a few potential leak cards from MH3 (the 4 flip walkers, in addition to Ajani).

https://www.reddit.com/r/MTGRumors/comments/1c6y81p/mh3_flipwalkers/

I wanted to work on at least some of them, as they seemed legit enough, and I have an use for them in my cube.

Is anyone against the temporary "leak custom set" introduced here? I'll clean it up once we know for sure it contains real or fake cards when the preview for the set will be there.

The custom status makes it opt in in the meantime.

### CODE REVIEW

File: Mage.Sets/src/mage/cards/r/RalLeylineProdigy.java [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+        this.supertype.add(SuperType.LEGENDARY);
+        this.subtype.add(SubType.RAL);
+        this.setStartingLoyalty(2);
+
+        this.color.setBlue(true);
+        this.color.setRed(true);
+        this.nightCard = true;
+
+        // Ral, Leyline Prodigy enters the battlefield with an additional loyalty counter on him for each instant and sorcery spell you've cast this turn.
+        this.addAbility(new EntersBattlefieldAbility(new RalLeylineProdigyCounterEffect()));
```
Comment (JayDi85) [inline Mage.Sets/src/mage/cards/r/RalLeylineProdigy.java:62]: Why not a standard effect here? Usage of custom effects/abilities should be kept to a minimum.

IMG_0219

Comment (Susucre) [reply Mage.Sets/src/mage/cards/r/RalLeylineProdigy.java:62]: Well I need a custom DynamicValue then.

Comment (JayDi85) [reply Mage.Sets/src/mage/cards/r/RalLeylineProdigy.java]: Thats correct usage — standard effects/abilities with customizable options like filter, target, condition.

If you can’t find compatible effect then extract it from same card to standard class. If it do not exists at all then create custom in card class.

Comment (JayDi85) [reply Mage.Sets/src/mage/cards/r/RalLeylineProdigy.java]: And next hint: if you use dynamic value then card hint must be added in most use cases :)

**Comment by Susucre** [comment]
I'm going to merge those 4 as custom cards like proposed in the PR.
The set is not that far, so we will have confirmation if they are real in a month or so.
Once we do, I'll clean up (remove) the custom set, and keep the cards if they are real.

**Review by JayDi85 — CHANGES_REQUESTED** [review]
@Susucre please, try to replace some effects and abilities by standard classes/code.

For info:
@xenohedron many cards in card PRs can be implemented by standard effects/abilities without custom code. It must be a priority. IDE regexp search can be useful to find same abilities/effects (replace dynamic text by `.+`). Example: 
shot_240420_225840

If new card with standard classes catch some bugs then it's ok to implement it and merge with that bugs. Bug fix of standard classes must be out of scope card implementation (well, only critical bugs like game freeze/crashes can stop from merge).

Less code -- better implementation.

File: Mage.Sets/src/mage/cards/t/TamiyoSeasonedScholar.java [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+        this.supertype.add(SuperType.LEGENDARY);
+        this.subtype.add(SubType.TAMIYO);
+        this.setStartingLoyalty(2);
+
+        this.color.setGreen(true);
+        this.color.setBlue(true);
+        this.nightCard = true;
+
+        // +2: Until your next turn, whenever a creature attacks you or a planeswalker you control, it gets -1/-0 until end of turn.
+        this.addAbility(new LoyaltyAbility(new TamiyoSeasonedScholarPlus2Effect(), 2));
```
[the remark this answers is not in the document]
Comment (Susucre) [reply Mage.Sets/src/mage/cards/t/TamiyoSeasonedScholar.java:46]: No, there is an issue with putting that delayed trigger and having it "until your next turn". I don't want to fix that bug here, so I have not used that common Effect here.

File: Mage.Sets/src/mage/sets/ModernHorizons3Leaks.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+ */
+public final class ModernHorizons3Leaks extends ExpansionSet {
+
+    private static final ModernHorizons3Leaks instance = new ModernHorizons3Leaks();
+
+    public static ModernHorizons3Leaks getInstance() {
+        return instance;
+    }
+
+    private ModernHorizons3Leaks() {
```
[the remark this answers is not in the document]
Comment (Susucre) [reply Mage.Sets/src/mage/sets/ModernHorizons3Leaks.java:24]: The cards are not officially in the set, their existence is not confirmed yet.
Custom set was the elegant way to test the cards without people adding to their decks thinking the cards were officials.

I'll just make a diverging fork at this point. And also with the various fixes you find smell bad.

File: Mage.Sets/src/mage/cards/t/TamiyoSeasonedScholar.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (xenohedron) [reply Mage.Sets/src/mage/cards/t/TamiyoSeasonedScholar.java:46]: See #11634 for context

File: Mage.Sets/src/mage/sets/ModernHorizons3Leaks.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (xenohedron) [reply Mage.Sets/src/mage/sets/ModernHorizons3Leaks.java:24]: I don't have a problem with making a temporary custom set for cards that are not an official preview (although question - how are images being supported?). It seems better than adding to the main set. (Personally, I'd just wait and not implement until after confirmation.)

[the remark this answers is not in the document]
Comment (JayDi85) [reply Mage.Sets/src/mage/sets/ModernHorizons3Leaks.java:24]: There have already been use cases where scryfall added preview cards with wrong abilities (e.g. due to translation errors).

[the remark this answers is not in the document]
Comment (xenohedron) [reply Mage.Sets/src/mage/sets/ModernHorizons3Leaks.java:24]: Sure. But a temporary translation error on scryfall is completely different from unverified card source that may not even be official card at all.

**Comment by xenohedron** [comment]
> For info: @ xenohedron many cards in card PRs can be implemented by standard effects/abilities without custom code. It must be a priority. IDE regexp search can be useful to find same abilities/effects (replace dynamic text by `.+`)

I agree. I review a lot of PRs for new card implementations, and I'm always pointing out where I see potential for using a common class rather than a custom class. However, sometimes there are subtle reasons why a custom class is a sensible tradeoff. I haven't reviewed this PR in detail to judge one way or the other.

> If new card with standard classes catch some bugs then it's ok to implement it and merge with that bugs. Bug fix of standard classes must be out of scope card implementation (well, only critical bugs like game freeze/crashes can stop from merge).

Well, it depends on the bug. If it's existing reported issue, or if only a discrepancy in rare circumstances, then sure, it shouldn't always block merge of new card implementation using common class. But in general it's good to use new card implementations as an inspiration to fix bugs that are noticed in the process.

**Comment by Susucre** [comment]
Alright, no custom leak in beta seems the consensus.
The cards are working in alpha if anyone care for them.

**Pull request closed without merging**